core/include/scipp/core/element/comparison.h

```diff
@@ -35,8 +35,9 @@ constexpr auto is_approx = overloaded{
 
 struct comparison_types_t {
   constexpr void operator()() const noexcept;
-  using types = decltype(std::tuple_cat(std::declval<arithmetic_type_pairs>(),
-                                        std::tuple<bool>{}));
+  using types = decltype(std::tuple_cat(
+      std::declval<arithmetic_type_pairs>(), std::tuple<bool>{},
+      std::tuple<core::time_point, core::time_point, core::time_point>{}));
```

Comment:
```suggestion
      std::tuple<core::time_point>{}));
```
should do

Reply:
Actually, is this better? `std::tuple<bool, core::time_point, core::time_point>`

Reply:
Why three arguments? Comparisons have two arguments. The output type (`bool`) is determined by `transform`.

Reply:
Aren't there cases where you specify it?

Reply:
Only for `transform_in_place`, not for `transform`.